app/(auth)/signup/_components/signupForm.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { Controller, FieldValues, useForm } from 'react-hook-form';
+
+import { AxiosError } from 'axios';
+import { useRouter } from 'next/navigation';
+
+import Button from '@/components/button/button';
+import Input from '@/components/input/input';
+import { postSignUpInfo } from '@/services/api';
+
+import MemberType from './memberType';
+
+const EMAIL_REGEX = /^[A-Za-z0-9]([-_.]?[A-Za-z0-9])*@[a-z0-9]([-_.]?[A-Za-z0-9])*\.[A-Za-z]{2,3}$/;
+const PASSWORD_REGEX = /^.{8,}$/;
+
+export default function SignUpForm() {
+  const router = useRouter();
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FieldValues>({ mode: 'all', defaultValues: { type: 'employee' } });
+
+  const FORM_FIELDS = [
+    {
+      name: 'email',
+      label: '이메일',
+      type: 'text',
+      placeholder: '이메일을 입력해주세요.',
+      validation: {
+        required: '이메일을 입력해주세요.',
+        pattern: {
+          value: EMAIL_REGEX,
+          message: '올바른 이메일 주소가 아닙니다.',
+        },
+      },
+    },
+    {
+      name: 'password',
+      label: '비밀번호',
+      type: 'password',
+      placeholder: '비밀번호를 입력해주세요.',
+      validation: {
+        required: '비밀번호를 입력해주세요.',
+        pattern: {
+          value: PASSWORD_REGEX,
+          message: '비밀번호를 8자 이상 입력해주세요.',
+        },
+      },
+    },
+    {
+      name: 'password_repeat',
```

Comment:
일반적으로 비밀번호 확인은 confirm으로 표현을 해줘서 아래와 같이 작성되면 좋을 것 같아요!
```suggestion
      name: 'confirmPassword',
```